src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -364,8 +361,26 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                                 Namespace = nspace,
                                                 Name = classDec.Identifier.ToString() + classDec.TypeParameterList,
                                                 Constraints = classDec.ConstraintClauses.ToString(),
+                                                ParentClass = null,
                                             };
 
+                                            LoggerClass currentLoggerClass = lc;
+                                            var parentLoggerClass = (classDec.Parent as ClassDeclarationSyntax);
```

Comment:
Does `ClassDeclarationSyntax` include `struct` or `record`?

Reply:
No I think we need `StructDeclarationSyntax` for that

Reply:
New commit adds support for logger methods nested inside structs, classes and records